miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/PoolRestController.java

```diff
@@ -279,6 +284,42 @@ public PoolPickerResponse getPickersBySearch(@RequestParam("platform") String pl
         PaginationFilter.platformType(PlatformType.valueOf(platform)));
   }
 
+  @RequestMapping(value = "avginsertsizes", method = RequestMethod.POST, produces = "application/json")
+  @ResponseBody
+  public JSONObject calculateAverageInsertSizes(@RequestBody JSONObject json) throws IOException {
+    String idString = json.getString("ids");
+    String[] split = idString.split(",");
+    List<Long> ids = new ArrayList<>();
+    for (int i = 0; i < split.length; i++) {
+      ids.add(Long.parseLong(split[i]));
+    }
+    Collection<Pool> pools = new ArrayList<>();
+    pools = poolService.listByIdList(ids);
+    List<String> avgs = new ArrayList<>();
+    for (Pool p : pools) {
+      long sum = 0;
+      Set<PoolableElementView> pevs = p.getPoolableElementViews();
+      if (pevs.size() > 0) {
+        long numpevs = 0;
+        for (PoolableElementView pev : pevs) {
+          if (pev != null && pev.getLibraryDnaSize() != null) {
+            sum += pev.getLibraryDnaSize();
+            numpevs++;
+          }
+        }
+        long avg = sum / numpevs;
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Make sure "numpevs" can't be zero before doing this division. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3518)